### PR TITLE
EZP-27670: Move a content item using action bar

### PR DESCRIPTION
**Pull request by yannickroger**
Link: https://jira.ez.no/browse/EZP-27670

## Description
Enables to move an item thanks to the Move button in the action bar.

## Tests
Manual tests and phpspec

### CODE REVIEW

File: src/bundle/Resources/views/locationview.html.twig [hunk 1]
```diff
@@ -89,6 +89,28 @@
...
+                })
+            }}
+                {{ form_widget(moveLocationForm.newParentLocationId) }}
+
+                {{
+                    form_widget(moveLocationForm.move, {
+                        'label': 'actionbar.move'|trans()|desc('Move'),
+                        'attr': {
+                            'class': 'ez-button ez-button-action ez-js-run-universal-discovery',
+                            'data-ud-title': 'actionbar.move.selectbox.title'|trans()|desc('Select the location where you want to move the location'),
```
Comment (SylvainGuittard) [inline src/bundle/Resources/views/locationview.html.twig]: Title: 
`Select a new location to move your content item`

File: src/bundle/Resources/views/locationview.html.twig [hunk 2]
```diff
@@ -89,6 +89,28 @@
...
+                            'data-ud-container': true,
+                            'data-ud-confirm-fill': '#' ~ moveLocationForm.newParentLocationId.vars.id,
+                            'data-ud-confirm-fill-with': 'location.id',
```
Comment (SylvainGuittard) [inline src/bundle/Resources/views/locationview.html.twig]: Could you please add `'data-ud-confirm-label': 'Move'`?
reference: https://github.com/ezsystems/hybrid-platform-ui/pull/95/files#diff-89cf99eb2ca64e40720b4f0294ec5407R45

**Comment by yannickroger** [comment]
@SylvainGuittard I took care of your feedback, also I added the disabled for root location as asked in the issue.

File: src/lib/Repository/UiLocationService.php [hunk 3]
```diff
@@ -115,6 +115,18 @@ public function trashLocationAndReturnParent(Location $location)
...
     /**
+     * Checks if a given location is root location.
+     *
+     * @param Location $location
+     *
+     * @return bool true if the location is root
+     */
+    private function isRootLocation(Location $location)
+    {
+        return $location->contentId == 1;
```
Comment (dpobel) [inline src/lib/Repository/UiLocationService.php]: In v1, Locations with a depth of 1 ie `/`, `/Media`, `/Users` (and `/Setup`) can not be moved as well as the *virtual root Location*, so I guess, the same should be implemented meaning you should probably test against the Location depth.

File: src/lib/Repository/UiLocationService.php [hunk 4]
```diff
@@ -123,7 +135,7 @@ public function trashLocationAndReturnParent(Location $location)
      */
     public function canRemoveLocation(Location $location)
     {
-        if ($location->contentId == 1) {
+        if ($this->isRootLocation($location)) {
```
Comment (dpobel) [inline src/lib/Repository/UiLocationService.php:138]: you could just write:
```php
return !$this->isRootLocation($location);
```

Comment (dpobel) [reply src/lib/Repository/UiLocationService.php:138]: and same in `canMoveLocation`

**Review by dpobel — APPROVED** [review]
besides the logic remark around *root* Location(s), +1

File: src/lib/Repository/UiLocationService.php (hunk 4 above)
Comment (yannickroger) [reply src/lib/Repository/UiLocationService.php:138]: but how will it do the permission check that comes afterwards?

Comment (dpobel) [reply src/lib/Repository/UiLocationService.php:138]: ah yeah missed this one, so this comment only applies to `canMoveLocation` :)

**Pull request merged**